plan/validator.go

```diff
@@ -302,3 +306,30 @@ func checkDuplicateColumnName(indexColNames []*ast.IndexColName) error {
 	}
 	return nil
 }
+
+// checkFieldLengthLimitation check the maximum length of the column
+// See https://dev.mysql.com/doc/refman/5.7/en/storage-requirements.html
+func checkFieldLengthLimitation(colDef *ast.ColumnDef) error {
+	tp := colDef.Tp
+	if tp == nil {
+		return nil
+	}
+	if tp.Flen > 4294967295 {
+		return errors.Errorf("Display width out of range for column '%s' (max = 4294967295)", colDef.Name.Name.O)
+	}
+	switch tp.Tp {
+	case mysql.TypeString:
```

Comment:
Varchar need to be based on the byte occupied by the character set to get the maximum length, I need to find out how to get the current field of character set.I tried to look for it.